webdav.go

```diff
@@ -149,8 +208,35 @@ type emptyBodyResponseWriter struct{ http.ResponseWriter }
 
 func (w emptyBodyResponseWriter) Write(data []byte) (int, error) { return 0, nil }
 
+// getFileResponseWriter is a helper to unpack embedded webdav.File(actually *os.File) in WebDAVFileWrapper
+type getFileResponseWriter struct{ http.ResponseWriter }
+
+func (w getFileResponseWriter) ReadFrom(r io.Reader) (int64, error) {
+	// check ReaderFrom
+	rr, _ := w.ResponseWriter.(io.ReaderFrom)
+	// *io.LimitedReader is used by http.ServeContent
+	if lr, ok := r.(*io.LimitedReader); ok {
+		// promote the embedded *os.File
+		if df, ok := lr.R.(*WebDAVFileWrapper); ok {
+			if df != nil {
+				lr.R = df.File
+			}
+		}
+	}
+	if rr != nil {
+		return rr.ReadFrom(r)
+	}
+	return io.Copy(w.ResponseWriter, r)
+}
+
 // Interface guards
 var (
 	_ caddyhttp.MiddlewareHandler = (*WebDAV)(nil)
 	_ caddyfile.Unmarshaler       = (*WebDAV)(nil)
+	_ webdav.ETager               = (*WebDAVFileInfo)(nil)
+	_ webdav.File                 = (*WebDAVFileWrapper)(nil)
+	_ webdav.FileSystem           = (*WebDAVDirFileSystemWrapper)(nil)
+
+	// To utilize optimizations like sendfile, *os.File has to be extracted from its wrapper
```

Comment:
Maybe move this comment, or repeat it, up by the definition of getFileResponseWriter, like, why is it necessary?